examples/async.rs

```diff
@@ -1,16 +1,25 @@
 extern crate futures;
 extern crate reqwest;
 extern crate tokio_core;
+extern crate tokio_io;
 
+use std::io;
 use futures::Future;
 
 fn main() {
     let mut core = tokio_core::reactor::Core::new().unwrap();
     let client = reqwest::unstable::async::Client::new(&core.handle()).unwrap();
 
-    let work = client.get("https://hyper.rs").unwrap().send().map(|res| {
-        println!("{}", res.status());
-    });
+    let work = client.get("http://localhost:8080/range/4096?duration=5&chunk_size=300")
```

Comment:
Idea that might make testing this as you go easier.

Not sure how easy it is to set up the [test stub server](https://github.com/seanmonstar/reqwest/blob/master/tests/support/server.rs#L107-L119) ([usage here](https://github.com/seanmonstar/reqwest/blob/master/tests/gzip.rs#L28-L37)) to chunk and *stutter* the packets back to behave like more like a scenario you'd want to test handling using async, but if its not too difficult I think it would make debugging and testing as you go easier than using thee example. At least that was my experience when I initially added gzip decoding. Also means you won't have to do that work at the end :smile: